tasks/installation.yml

```diff
@@ -156,13 +120,41 @@
     gpgkey: https://packages.grafana.com/gpg.key
     sslverify: true
     sslcacert: /etc/pki/tls/certs/ca-bundle.crt
-  when: icinga2_web_grafana.enabled
+  when:
+    - icinga2_web_grafana.enabled
+    - ansible_distribution == 'RedHat'
 
 - name: Install grafana
-  ansible.builtin.yum:
+  ansible.builtin.dnf:
     name: grafana
     state: present
-  when: icinga2_web_grafana.enabled
+  when:
+    - icinga2_web_grafana.enabled
+    - ansible_distribution == 'RedHat'
+
+- name: Add grafana apt key
+  ansible.builtin.apt_key:
+    id: 963FA27710458545
+    url: https://apt.grafana.com/gpg.key
+    state: present
+  when:
+    - icinga2_web_grafana.enabled
+    - ansible_distribution == 'Debian'
+
+- name: Add grafana repository
+  ansible.builtin.apt_repository:
+    repo: deb https://apt.grafana.com stable main
+    state: present
+  when:
+    - icinga2_web_grafana.enabled
+    - ansible_distribution == 'Debian'
+
+- name: Install grafana
+  ansible.builtin.apt:
+    name: grafana
+  when:
+    - icinga2_web_grafana.enabled
+    - ansible_distribution == 'Debian'
```

Comment:
```suggestion
- name: Configure grafana for debian
  when:
    - icinga2_web_grafana.enabled
    - ansible_distribution == 'Debian'
  block:
    - name: Add grafana apt key
      ansible.builtin.apt_key:
        id: 963FA27710458545
        url: https://apt.grafana.com/gpg.key
        state: present
    
    - name: Add grafana repository
      ansible.builtin.apt_repository:
        repo: deb https://apt.grafana.com stable main
        state: present
    
    - name: Install grafana
      ansible.builtin.apt:
        name: grafana
```

Reply:
If multiple tasks use the same when condition one after the other, I would put them in a block. But that's my personal preference, up to you if you want to change it.

Reply:
Afaik, we do not really use this in our roles yet, best-practice wise; I know it exists but haven't really used it in any of our roles.